src/Application/Console/Formatters/PathShortener.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\PhpInsights\Application\Console\Formatters;
+
+use NunoMaduro\PhpInsights\Domain\Details;
+
+/**
+ * @internal
+ */
+final class PathShortener
+{
+    /**
+     * @param array<string> $paths
+     *
+     * @return string
+     */
+    public static function extractCommonPath(array $paths): string
```

Comment:
My solution was more something like this. Don't know if you find my solution simpler or not :)

```php
        $paths = array_values($paths);
        sort($paths);

        $first = $paths[0];
        $last = $paths[count($paths)-1];
        $length = min(strlen($first), strlen($last));

        for ($index=0; $index<$length && $first[$index]==$last[$index]; $index++);

        $prefix = mb_substr($first, 0, $index ?? 0);

        return mb_substr(
            $prefix,
            0,
            mb_strrpos($prefix, DIRECTORY_SEPARATOR)+1
        );
```

Reply:
@olivernybroe Yes your solution is simpler and I like it more. Thanks a lot :)